### PR TITLE
docs(samples): add create_migration_workflow snippet

### PR DESCRIPTION
**Pull request by steffnay**
Adds snippet and testing structure.

 🦕

### CODE REVIEW

**Comment by dandhlee** [comment]
Created cl/449903650 to enable tests for Samples.

File: samples/snippets/create_migration_workflow.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START bigquery_migration_create_workflow]
+def create_migration_workflow(gcs_input_path, gcs_output_path, project_id):
+    """Creates a migration workflow of a Batch SQL Translation and prints the response."""
+
+    from google.cloud import bigquery_migration_v2
+
+    parent = "projects/%s/locations/us" % project_id
```
Comment (parthea) [inline samples/snippets/create_migration_workflow.py:21]:
```suggestion
    parent = f"projects/{project_id}/locations/us"
```

File: samples/snippets/create_migration_workflow.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    # Construct a BigQuery Migration client object.
+    client = bigquery_migration_v2.MigrationServiceClient()
+
+    # Set the source dialect to Teradata SQL.
+    source_dialect = bigquery_migration_v2.Dialect()
+    source_dialect.teradata_dialect = bigquery_migration_v2.TeradataDialect(
+        mode=bigquery_migration_v2.TeradataDialect.Mode.SQL
+    )
+
+    # Set the target dialect to Bigquery dialect.
```
Comment (parthea) [inline samples/snippets/create_migration_workflow.py:32]:
```suggestion
    # Set the target dialect to BigQuery dialect.
```

File: samples/snippets/requirements-test.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+pytest==6.2.5
+google-cloud-testutils==1.3.0
+google-api-core==2.0.1
+google-cloud-storage
```
Comment (parthea) [inline samples/snippets/requirements-test.txt:4]: Please pin to a specific version

File: samples/snippets/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+google-cloud-bigquery-migration
```
Comment (parthea) [inline samples/snippets/requirements.txt:1]: Please pin to a specific version

**Review by parthea — APPROVED** [review]

**Pull request merged**